compatibility-api/src/main/scala/za/co/absa/hyperdrive/compatibility/api/CompatibleDeltaCDCToSnapshotWriter.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.compatibility.api
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.streaming.{StreamingQuery, Trigger}
+
+trait CompatibleDeltaCDCToSnapshotWriter {
+  def write(dataFrame: DataFrame): StreamingQuery
+}
+
+case class DeltaCDCToSnapshotWriterConfiguration(
+  checkpointLocationPropName: String,
```

Comment:
Because in Parquet writer it is taken from StreamWriterProperties.CheckpointLocation but that is not available in compatibility_spark-X package so I still use StreamWriterProperties.CheckpointLocation but I pass it as param from ingestor-default package

Reply:
Hmm, but this is not a great solution, imo. It took me a while to grasp that it's just a constant. You could add the module as a dependency or simply hard-code "checkpointLocation"

Reply:
Done